src/Workspaces/Core/Portable/CodeActions/Operations/CodeActionOperation.cs

```diff
@@ -25,10 +27,10 @@ public virtual void Apply(Workspace workspace, CancellationToken cancellationTok
         {
         }
 
-        internal virtual bool TryApply(Workspace workspace, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        internal virtual Task<bool> TryApplyAsync(Workspace workspace, IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
             this.Apply(workspace, cancellationToken);
```

Comment:
❗ This needs to switch to the main thread before calling `Apply`. The synchronous method is a public API with a documented UI thread guarantee.

Reply:
true.  do you have a good suggestion on how to do that here given that we're at teh Workspaces layer?

Reply:
i could also doc this method as being called on the UI thread, and then ensure the one caller does this (similar to how Apply works), is that ok with you?

Reply:
Asynchronous methods are callable from any context. If the implementation requires a specific context, it is always allowed to switch to that context at the point it is required.

Deviating from this standard practice makes the code more difficult to consume, and should be avoided to the degree possible.

Reply:
Ok.  My point is simply:. How can we accomplish that switch at this level?  Thanks! :-)

Reply:
specifically, this is at the workspace layer, which doesn't have access to IThreadingContext.  So i'm not sure how to updat ethis so that this code can switch to UI thread prior to calling Apply.